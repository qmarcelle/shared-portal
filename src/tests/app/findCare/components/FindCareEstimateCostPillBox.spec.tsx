import FindCare from '@/app/findcare';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(<FindCare />);
};
// Mock useRouter:
const mockPush = jest.fn();
jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      prefetch: () => null,
      push: mockPush,
    };
  },
}));
jest.mock('src/auth', () => ({
  auth: jest.fn(),
}));
const vRules = {
  user: {
    currUsr: {
      plan: {
        memCk: '123456',
      },
    },
    vRules: {
      showPharmacyTab: true,
      terminated: false,
      wellnessOnly: false,
      fsaOnly: false,
      vision: true,
      futureEffective: false,
      katieBeckNoBenefitsElig: false,
    },
  },
};

process.env.NEXT_PUBLIC_IDP_PROVIDER_DIRECTORY =
  'https://uat.bcbst.sapphirecareselect.com/auth/metadata.xml';
process.env.NEXT_PUBLIC_PROVIDER_DIRECTORY_VITALS_SSO_TARGET =
  'https://uat.bcbst.sapphirecareselect.com/{DEEPLINK}';
process.env.NEXT_PUBLIC_IDP_CVS_CAREMARK = 'SP_CVS_BCBSTN';
process.env.NEXT_PUBLIC_CVS_SSO_TARGET =
  'https://www.caremark.com/{DEEPLINK}?newLogin=yes';
process.env.NEXT_PUBLIC_IDP_EYEMED = 'EYEMED_ITTEST';
process.env.NEXT_PUBLIC_EYEMED_SSO_TARGET =
  'https://member.eyemedvisioncare.com/bcbst/en/private/{DEEPLINK}';

describe('FindCarePillBox', () => {
  it('should render UI correctly', () => {
    const component = renderUI();

    screen.getByRole('heading', {
      name: 'Planning for a procedure? Estimate costs for:',
    });
    fireEvent.click(screen.getByLabelText('Medical'));

    expect(mockPush).toHaveBeenCalledWith(
      '/sso/launch?PartnerSpId=https://uat.bcbst.sapphirecareselect.com/auth/metadata.xml&TargetResource=https://uat.bcbst.sapphirecareselect.com/?guided_search=wayfinding_home_findCost_header',
    );

    fireEvent.click(screen.getByLabelText('Dental'));
    expect(mockPush).toHaveBeenCalledWith('/member/findcare/dentalcosts');

    expect(component.baseElement).toMatchSnapshot();
  });
  it('should not call sso if Pharamcybenifits visibility rule false on click of prescription drugs', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    vRules.user.vRules.showPharmacyTab = false;
    mockAuth.mockResolvedValueOnce(vRules);
    const component = await renderUI();
    fireEvent.click(screen.getByLabelText('Prescription Drugs'));
    expect(mockPush).not.toHaveBeenCalledWith(
      '/sso/launch?PartnerSpId=SP_CVS_BCBSTN&TargetResource=https://www.caremark.com/drugSearchInit.do?newLogin=yes',
    );
    expect(component.baseElement).toMatchSnapshot();
  });

  it('should not call sso if visionBenefits visibility rule true on click of prescription drugs', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    vRules.user.vRules.vision = false;
    mockAuth.mockResolvedValueOnce(vRules);
    const component = await renderUI();
    fireEvent.click(screen.getByLabelText('Vision'));
    expect(mockPush).not.toHaveBeenCalledWith(
      '/sso/launch?PartnerSpId=EYEMED_ITTEST&TargetResource=https://member.eyemedvisioncare.com/bcbst/en/private/know-before-you-go',
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
