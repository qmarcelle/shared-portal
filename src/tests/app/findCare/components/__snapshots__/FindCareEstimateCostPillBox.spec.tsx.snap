// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindCarePillBox should not call sso if Pharamcybenifits visibility rule false on click of prescription drugs 1`] = `
<body>
  <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col app-content app-base-font-color md:p-0 p-4"
      >
        <section
          class="card-main max-sm:mt-24 mt-8 "
        >
          <div
            class="flex flex-row items-center input   "
          >
            <input
              class="body-1 search-textfield"
              name="search-Field"
              placeholder="Search by procedure, specialty, facility or provider name..."
              type="text"
              value=""
            />
            <div
              class="flex flex-col items-end"
            >
              <button
                aria-label="Search"
                class="flex flex-col p-[0.125rem]  "
                type="submit"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="icon searchhover "
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </button>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for care? Find a:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Primary Care Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Primary Care Provider
                  </button>
                  <button
                    aria-label="Dentist"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dentist
                  </button>
                  <button
                    aria-label="Mental Health Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Mental Health Provider
                  </button>
                  <button
                    aria-label="Eye Doctor"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Eye Doctor
                  </button>
                  <button
                    aria-label="Pharmacy"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Pharmacy
                  </button>
                  <button
                    aria-label="Virtual Care"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Virtual Care
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? Estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Medical"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Medical
                  </button>
                  <button
                    aria-label="Dental"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dental
                  </button>
                  <button
                    aria-label="Prescription Drugs"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Prescription Drugs
                  </button>
                  <button
                    aria-label="Vision"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Vision
                  </button>
                </div>
              </section>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body mt-4"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="flex flex-col  cursor-pointer"
            >
              <div
                class="flex flex-col flex flex-col"
              >
                <div
                  class="flex flex-row"
                >
                  <p
                    class="title-1"
                  >
                    View Care Options
                  </p>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="card-elevated mb-4"
                >
                  <div
                    class="flex flex-row align-top m-4 mt-8"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <img
                        alt="Primary Care"
                        class="max-md:w-[80px] max-md:h-[80px]"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <span
                      style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold primary-color"
                        style="display: block;"
                      >
                        Primary Care Options
                      </p>
                      <p
                        class="body-1 mt-2"
                        style="display: block;"
                      >
                        Learn more about Primary Care Providers and view your options.
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="card-elevated mb-4"
                >
                  <div
                    class="flex flex-row align-top m-4 mt-8"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <img
                        alt="Mental Care"
                        class="max-md:w-[80px] max-md:h-[80px]"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <span
                      style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold primary-color"
                        style="display: block;"
                      >
                        Mental Care Options
                      </p>
                      <p
                        class="body-1 mt-2"
                        style="display: block;"
                      >
                        Learn more about Mental Health Providers and view your options.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
    </main>
  </div>
</body>
`;

exports[`FindCarePillBox should not call sso if visionBenefits visibility rule true on click of prescription drugs 1`] = `
<body>
  <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col app-content app-base-font-color md:p-0 p-4"
      >
        <section
          class="card-main max-sm:mt-24 mt-8 "
        >
          <div
            class="flex flex-row items-center input   "
          >
            <input
              class="body-1 search-textfield"
              name="search-Field"
              placeholder="Search by procedure, specialty, facility or provider name..."
              type="text"
              value=""
            />
            <div
              class="flex flex-col items-end"
            >
              <button
                aria-label="Search"
                class="flex flex-col p-[0.125rem]  "
                type="submit"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="icon searchhover "
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </button>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for care? Find a:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Primary Care Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Primary Care Provider
                  </button>
                  <button
                    aria-label="Dentist"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dentist
                  </button>
                  <button
                    aria-label="Mental Health Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Mental Health Provider
                  </button>
                  <button
                    aria-label="Eye Doctor"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Eye Doctor
                  </button>
                  <button
                    aria-label="Pharmacy"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Pharmacy
                  </button>
                  <button
                    aria-label="Virtual Care"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Virtual Care
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? Estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Medical"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Medical
                  </button>
                  <button
                    aria-label="Dental"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dental
                  </button>
                  <button
                    aria-label="Prescription Drugs"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Prescription Drugs
                  </button>
                  <button
                    aria-label="Vision"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Vision
                  </button>
                </div>
              </section>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body mt-4"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="flex flex-col  cursor-pointer"
            >
              <div
                class="flex flex-col flex flex-col"
              >
                <div
                  class="flex flex-row"
                >
                  <p
                    class="title-1"
                  >
                    View Care Options
                  </p>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="card-elevated mb-4"
                >
                  <div
                    class="flex flex-row align-top m-4 mt-8"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <img
                        alt="Primary Care"
                        class="max-md:w-[80px] max-md:h-[80px]"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <span
                      style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold primary-color"
                        style="display: block;"
                      >
                        Primary Care Options
                      </p>
                      <p
                        class="body-1 mt-2"
                        style="display: block;"
                      >
                        Learn more about Primary Care Providers and view your options.
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="card-elevated mb-4"
                >
                  <div
                    class="flex flex-row align-top m-4 mt-8"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <img
                        alt="Mental Care"
                        class="max-md:w-[80px] max-md:h-[80px]"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <span
                      style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold primary-color"
                        style="display: block;"
                      >
                        Mental Care Options
                      </p>
                      <p
                        class="body-1 mt-2"
                        style="display: block;"
                      >
                        Learn more about Mental Health Providers and view your options.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
    </main>
  </div>
</body>
`;

exports[`FindCarePillBox should render UI correctly 1`] = `
<body>
  <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col app-content app-base-font-color md:p-0 p-4"
      >
        <section
          class="card-main max-sm:mt-24 mt-8 "
        >
          <div
            class="flex flex-row items-center input   "
          >
            <input
              class="body-1 search-textfield"
              name="search-Field"
              placeholder="Search by procedure, specialty, facility or provider name..."
              type="text"
              value=""
            />
            <div
              class="flex flex-col items-end"
            >
              <button
                aria-label="Search"
                class="flex flex-col p-[0.125rem]  "
                type="submit"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="icon searchhover "
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </button>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for care? Find a:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Primary Care Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Primary Care Provider
                  </button>
                  <button
                    aria-label="Dentist"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dentist
                  </button>
                  <button
                    aria-label="Mental Health Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Mental Health Provider
                  </button>
                  <button
                    aria-label="Eye Doctor"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Eye Doctor
                  </button>
                  <button
                    aria-label="Pharmacy"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Pharmacy
                  </button>
                  <button
                    aria-label="Virtual Care"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Virtual Care
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? Estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Medical"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Medical
                  </button>
                  <button
                    aria-label="Dental"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dental
                  </button>
                  <button
                    aria-label="Prescription Drugs"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Prescription Drugs
                  </button>
                  <button
                    aria-label="Vision"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Vision
                  </button>
                </div>
              </section>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body mt-4"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="flex flex-col  cursor-pointer"
            >
              <div
                class="flex flex-col flex flex-col"
              >
                <div
                  class="flex flex-row"
                >
                  <p
                    class="title-1"
                  >
                    View Care Options
                  </p>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="card-elevated mb-4"
                >
                  <div
                    class="flex flex-row align-top m-4 mt-8"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <img
                        alt="Primary Care"
                        class="max-md:w-[80px] max-md:h-[80px]"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <span
                      style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold primary-color"
                        style="display: block;"
                      >
                        Primary Care Options
                      </p>
                      <p
                        class="body-1 mt-2"
                        style="display: block;"
                      >
                        Learn more about Primary Care Providers and view your options.
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="card-elevated mb-4"
                >
                  <div
                    class="flex flex-row align-top m-4 mt-8"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <img
                        alt="Mental Care"
                        class="max-md:w-[80px] max-md:h-[80px]"
                        data-nimg="1"
                        decoding="async"
                        height="40"
                        loading="lazy"
                        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                        style="color: transparent;"
                        width="40"
                      />
                    </div>
                    <span
                      style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                    />
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold primary-color"
                        style="display: block;"
                      >
                        Mental Care Options
                      </p>
                      <p
                        class="body-1 mt-2"
                        style="display: block;"
                      >
                        Learn more about Mental Health Providers and view your options.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
    </main>
  </div>
</body>
`;
